propagate_env_vars.py

```diff
@@ -58,6 +59,11 @@ def main():
             for env_var in (" " + read_makefile_var("AARCH64_ENV") + " ").split(" -e ")
             if "=" in env_var
         ])
+    elif IS_OSX:
+        env.update({
+            "CFLAGS": "$CFLAGS -flto",
+            "LDFLAGS": "$LDFLAGS -arch x86_64",
+        })
```

Comment:
The `-flto` option should always be in the `CFLAGS` already.
```suggestion
        append_env(env, "LDFLAGS", "-arch x86_64")
```